### PR TITLE
Refactors README for mlmd_bench.

### PR DESCRIPTION
**Pull request by BrianSong**
Refactors README for mlmd_bench with more background introduction and adds benchmark coverage table.

### CODE REVIEW

File: ml_metadata/tools/mlmd_bench/README.md [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
-The performance metrics of interests include throughputs of concurrent
-operations and data sizes.
+mlmd_bench is a MLMD benchmark tool that can measure the MLMD query facility performance and scalability for different backends and deployment settings. The performance metrics of interests include throughputs of concurrent operations and data sizes.
 
-## HOW TO USE
+## Introduction
+
+MLMD exposes a provenance graph data model, which consists of {Artifact, Execution, Context, Type} as nodes, and {Event, Association, Attribution, Instance} as edges. Based on the data model, it defines a set of APIs, such as creating and updating nodes and edges, listing nodes by types, traversing nodes via edges.  
+
+The APIs are implemented on various backends (e.g., MySQL, Sqlite, CNS, Spanner) and deployed in different modes (e.g., library, grpc server). On the other hand, the integration of MLMD with different partners have different use cases and exhibit different read/write workloads. To improve API performance, it often requires proper optimization in MLMD query implementation or schema reorganization.
```
Comment (hughmiao) [inline ml_metadata/tools/mlmd_bench/README.md:9]: (e.g., MySQL, Sqlite)

File: ml_metadata/tools/mlmd_bench/README.md [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+mlmd_bench is a MLMD benchmark tool that can measure the MLMD query facility performance and scalability for different backends and deployment settings. The performance metrics of interests include throughputs of concurrent operations and data sizes.
 
-## HOW TO USE
+## Introduction
+
+MLMD exposes a provenance graph data model, which consists of {Artifact, Execution, Context, Type} as nodes, and {Event, Association, Attribution, Instance} as edges. Based on the data model, it defines a set of APIs, such as creating and updating nodes and edges, listing nodes by types, traversing nodes via edges.  
+
+The APIs are implemented on various backends (e.g., MySQL, Sqlite, CNS, Spanner) and deployed in different modes (e.g., library, grpc server). On the other hand, the integration of MLMD with different partners have different use cases and exhibit different read/write workloads. To improve API performance, it often requires proper optimization in MLMD query implementation or schema reorganization.    
+
+To guide the performance tuning of MLMD and better support users, we build a benchmarking tool, mlmd_bench, which can:
```
Comment (hughmiao) [inline ml_metadata/tools/mlmd_bench/README.md:11]: To better support users and guide the performance tuning of MLMD

File: ml_metadata/tools/mlmd_bench/README.md [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+| FillTypes      | PutArtifactType /<br> PutExecutionType /<br> PutContextType       | Insert / Update <br> Artifact Type / Execution Type / Context Type <br> Number of properties for each type |
+| FillNodes   | PutArtifact /<br> PutExecution /<br> PutContext        | Insert / Update<br>Artifact / Execution / Context<br>Number of properties for each node <br> Length for string properties of each node<br>APIs’ specification(e.g. number of nodes per request)|
+| FillContextEdges      | PutAttributionsAndAssociation       | Attribution / Association<br>Context / Non-context popularity<br>APIs’ specification(e.g. number of context edges per request)|
+| FillEvents      | PutEvent       | Input / Output Event<br>Artifact / Execution popularity<br>APIs’ specification(e.g. number of events per request)|
+| ReadTypes      | GetArtifactTypes /<br> GetArtifactTypesByID /<br> GetArtifactType ...| The type listing / querying APIs<br>APIs’ specification(e.g. number of ids per request)|
+| ReadNodesByProperties      | GetArtifactsByID /<br> GetArtifactsByType /<br> GetArtifactByTypeAndName ...| The nodes listing / querying APIs<br>APIs’ specification(e.g. number of ids per request)|
```
Comment (hughmiao) [inline ml_metadata/tools/mlmd_bench/README.md:23]: drop the '...' ? ditto elsewhere

Comment (BrianSong) [reply ml_metadata/tools/mlmd_bench/README.md]: Done!

File: ml_metadata/tools/mlmd_bench/README.md (hunk 2 above)
Comment (BrianSong) [reply ml_metadata/tools/mlmd_bench/README.md]: Good point. Done.

File: ml_metadata/tools/mlmd_bench/README.md (hunk 1 above)
Comment (BrianSong) [reply ml_metadata/tools/mlmd_bench/README.md]: Done.

**Pull request merged**